File: app/src/main/java/hegde/mahesh/avre/interpreter/SnippetEvalException.java
package hegde.mahesh.avre.interpreter;

import androidx.annotation.Nullable;

import java.util.Locale;

public class SnippetEvalException extends Exception {
    int line;
    int column;
    Throwable cause;

    public SnippetEvalException(int line, int column, Throwable cause) {
        super(cause.getMessage());
        this.line = line;
        this.column = column;
        this.cause = cause;
    }

    @Nullable
    @Override
    public String getMessage() {
        if (line == 0) {
            return cause.getMessage();
        }
        return String.format(Locale.getDefault(), "[line %s]: %s", line, cause.toString());
    }
}

File: app/src/main/java/hegde/mahesh/avre/interpreter/SnippetEvalException.kt
package hegde.mahesh.avre.interpreter

import java.util.*

class SnippetEvalException(private var line: Int, override var cause: Throwable) : Exception(cause.message) {
    override val message: String?
        get() = if (line == 0) {
            cause.message
        } else String.format(Locale.getDefault(), "[line %s]: %s", line, cause.toString())
}